lib/reek/configuration/default_directive.rb

```diff
@@ -21,8 +21,10 @@ module DefaultDirective
       # @return [self]
       def add(detectors_configuration)
         detectors_configuration.each do |name, configuration|
-          detector = key_to_smell_detector(name)
-          self[detector] = (self[detector] || {}).merge configuration
+          if configuration
```

Comment:
Empty configs should be filtered out earlier on. In fact, if you move the handling of the error, Reek will not run with nil configurations so this condition will no longer be needed.